ci/docker/conda-integration.dockerfile

```diff
@@ -25,32 +25,26 @@ ARG node=14
 ARG jdk=8
 ARG go=1.15
 
+# Uninstall unused space-consuming packages
+# (XXX: it would be better not to install them, but they are used by other
+#  builds which are also based on conda-cpp)
+RUN conda uninstall -q clangdev llvmdev valgrind
```

Comment:
Does this really help? Docker layers are on top of each other. This thus then only "hide" the files but does not remove them from disk.

Reply:
Ah... That's a good question. But I assume image layers are compressed? At least this addresses the uncompressed container size.

Reply:
Once you install something in a layer, it gets added to the total image size. This is why we should prefer install and cleanup up steps in the same layer.

Reply:
In this case, the install was done in conda-cpp...

Reply:
We should check whether we need the clangdev and llvmdev packages. I would guess that llvm would be sufficient as we link dynamically. Not sure what part of clang(dev) we need.

Reply:
Instead of focusing on shrinking the docker images, I'd rather consider to free up more space on the github actions hosted agents (if we can easily), see the space usage here: https://github.com/apache/arrow/pull/9814/checks?check_run_id=2205470543#step:4:83

The GHA hosted agents are provisioned using packer, here is the configuration for ubuntu 20.04: https://github.com/actions/virtual-environments/blob/main/images/linux/toolsets/toolset-2004.json

Presumably the android SDK consumes a lot of space, but there are other toolsets we don't use either.

Reply:
Well, we can do both. The size of docker images and container also affects local development.

Reply:
I meant deleting is probably easier and quicker, reduced image size would be better of course.

Reply:
Hmm, the problem is that the delete step can take 1 minute if I remove the Android toolkit...